AppController/djinn.rb

```diff
@@ -3511,21 +3511,21 @@ def start_soap_server
     MonitInterface.start(:uaserver, start_cmd, nil, env_vars)
   end
 
+  # Shadow is the only node to call this method, and is called upon
+  # startup.
   def start_datastore
     verbose = @options['verbose'].downcase == 'true'
-    db_proxy = nil
     db_nodes = []
     @state_change_lock.synchronize {
       @nodes.each { |node|
-        db_proxy = node.private_ip if node.is_load_balancer?
         db_nodes << node if node.is_db_master? || node.is_db_slave?
       }
     }
 
-    HelperFunctions.log_and_crash('db proxy ip was nil') if db_proxy.nil?
-
-    assignments = {}
+    # Assign the proper number of Datastore processes on each database
+    # machine.
     @nodes.each { |node|
```

Comment:
It should be `db_nodes.each { |node|`. Currently, if you look into [build logs](https://ocd.appscale.com:8080/job/Hawkeye/14686/artifact/logs-appscale-testing-d8721831-fef1-45d7-b356-5a6f1ea0d20d/symlinks/shadow/192.168.102.73/appscale/) you'll see that datastore was started on all nodes.